cmd/docker/docker.go

```diff
@@ -194,9 +197,48 @@ func tryPluginRun(dockerCli command.Cli, cmd *cobra.Command, subcommand string,
 	}
 	plugincmd.Env = append(envs, plugincmd.Env...)
 
+	// TODO(laurazard): this will break if we run this plugin more than once,
+	// as we will collide with the already-in-use socket name. We should retry
+	// a couple of times and add a counter or some other unique element to them
+	// socket name
+	pluginCloseSocketAddr := net.UnixAddr{Name: "@docker_cli_" + subcommand, Net: "unix"}
+	listener, err := net.ListenUnix("unix", &pluginCloseSocketAddr)
+	if err != nil {
+		return errors.New("bork")
+	}
+	defer listener.Close()
+	plugincmd.Env = append(envs, plugin.CliPluginCloseSocketEnvKey+"="+pluginCloseSocketAddr.Name)
```

Comment:
Could the CI failures be related to setting the `.Env` field? ISTR "empty" or "nil" means; keep the environment variables, but setting the field to any other value will discard the environment

Reply:
We already modify the environment image (which indeed, needs to be constructed from whole cloth if you modify it) above; this code just sets it to a new slice with a new `append()`.

Reply:
https://pkg.go.dev/os/exec#Cmd.Env

```go
// Env specifies the environment of the process.
	// Each entry is of the form "key=value".
	// If Env is nil, the new process uses the current process's
	// environment.
	// If Env contains duplicate environment keys, only the last
	// value in the slice for each duplicate key is used.
	// As a special case on Windows, SYSTEMROOT is always added if
	// missing and not explicitly set to the empty string.
	Env [][string](https://pkg.go.dev/builtin#string)
```


So possibly need to append to `os.Environ()` or `plugincmd.Environ()` https://pkg.go.dev/os/exec#example-Cmd.Environ

Reply:
> We already modify the environment image (which indeed, needs to be constructed from whole cloth if you modify it) above

Yes, was looking at that, but wondering if it _actually_ set anything (if `env` is nil, it would potentially still be nil(ish?)

Reply:
It does (and appending to a nil slice works too!), but maybe that's not enough everywhere? It works on my ~~machine~~ Ubuntu VM